dotcom-rendering/src/apps/server/index.ts

```diff
@@ -1,11 +1,28 @@
 import { RequestHandler } from 'express';
+import { enhanceArticleType } from '../../web/server';
+import { articleToHtml } from './articleToHtml';
 
 const getStack = (e: unknown): string =>
 	e instanceof Error ? e.stack ?? 'No error stack' : 'Unknown error';
 
-export const handleAppsArticle: RequestHandler = (_req, res) => {
+/**
+ * Formats script paths as a Link header
+ * @see https://datatracker.ietf.org/doc/html/rfc5988#section-5.5
+ * @param scriptPaths - the script paths to include in the Link header
+ */
+const makePrefetchHeader = (scriptPaths: string[]): string =>
+	scriptPaths.reduce(
+		(acc, scriptPath) => acc + `<${scriptPath}>; rel=prefetch,`,
+		'',
+	);
+
+export const handleAppsArticle: RequestHandler = ({ body }, res) => {
 	try {
-		res.status(200).send('<pre>Not supported</pre>');
+		const article = enhanceArticleType(body);
+		const { html, clientScripts } = articleToHtml(article);
+
+		// The Android app will cache these assets to enable offline reading
```

Comment:
Out of curiosity, why is this specific to Android? Does iOS not use this?

Reply:
as far as I could tell, iOS and Android use different mechanisms to support offline reading. However, I'm not sure why this difference exists, or whether they could both use the same mechanism. It might be worth reaching out to the apps teams to double check at some point.

**iOS**

The AR build process generates an HTML file which references all the static assets like fonts and client scripts (https://github.com/guardian/dotcom-rendering/blob/main/apps-rendering/webpack.config.ts#L231). iOS [fetches this file](https://github.com/guardian/ios-live/blob/7f4a8c158f0487554e27fcb6297ab7ce11032615/GLA/GLA/Classes/AppStart.swift#L97) in the app start lifecycle and caches the assets.

**Android**

The Android app [intercepts requests for fonts](https://github.com/guardian/android-news-app/blob/5c4cc2eaa344ca2e81d6de485c8f8f04474b433c/webview/src/main/java/com/theguardian/webview/http/FontInterceptor.kt#L11), serving them with assets from the app bundle.

For client scripts, it relies on the `Link` header in the response headers to determine which assets need to be cached.

Reply:
> an HTML file which references all the static assets like fonts and client scripts

Are you sure that file references the client script and not just the fonts?

Reply:
I'm super interested by how iOS does this, I'm guessing this is something we'll need to add to the apps webpack build? Happy to pair on it if so!